util/angleoptimize.py

```diff
@@ -0,0 +1,25 @@
+from math import tau, floor
+from wpimath.geometry import Rotation2d
+
+
+def optimizeAngle(currentAngle: Rotation2d, targetAngle: Rotation2d) -> Rotation2d:
+    currentAngle = currentAngle.radians()
+
+    closestFullRotation = (
+        floor(abs(currentAngle / tau)) * (-1 if currentAngle < 0 else 1) * tau
```

Comment:
Don't really understand this so imma just trust that it's right

Reply:
This was tested from the drive subsystem optimization, I moved it over because it wasn’t used just in the drive anymore